features/eolearn/features/clustering.py

```diff
@@ -0,0 +1,145 @@
+            n_clusters must be None nd compute_full_tree must be True
+        :type distance_threshold: float or None
+        :param n_clusters: The number of clusters found by the algorithm. If distance_threshold=None, it will be equal
+            to the given n_clusters
+        :type n_clusters: int or None
+        :param affinity: Metric used to compute the linkage. Can be “euclidean”, “l1”, “l2”, “manhattan”, “cosine”.
+        :type affinity: str
+        :param linkage: Which linkage criterion to use. The linkage criterion determines which distance to use between
+            sets of observation. The algorithm will merge the pairs of cluster that minimize this criterion.
+            - ward minimizes the variance of the clusters being merged.
+            - average uses the average of the distances of each observation of the two sets.
+            - complete or maximum linkage uses the maximum distances between all observations of the two sets.
+            - single uses the minimum of the distances between all observations of the two sets.
+        :type linkage: {“ward”, “complete”, “average”, “single”}
+        :param remove_small: If greater than 0, removes all clusters that have less points as "remove_small"
+        :type remove_small: int
+        :param connectivity: Connectivity matrix. Defines for each sample the neighboring samples following a given
+            structure of the data. This can be a connectivity matrix itself or a callable that transforms the data into
+            a connectivity matrix, such as derived from kneighbors_graph. If set to None it uses the graph that has
+            adjacent pixels connected.
+        :type connectivity: array-like, callable or None
+        :param mask_name: An optional mask feature used for exclusion of the area from clustering
+        :type mask_name: str
+        """
+        self.features = features
+        self.distance_threshold = distance_threshold
+        self.affinity = affinity
+        self.linkage = linkage
+        self.new_feature_name = new_feature_name
+        self.n_clusters = n_clusters
+        self.compute_full_tree = 'auto'
+        if distance_threshold is not None:
+            self.compute_full_tree = True
+        if remove_small < 0:
+            raise ValueError('remove_small argument should be non-negative')
+        self.remove_small = remove_small
+        self.connectivity = connectivity
+        self.mask_name = mask_name
+
+    @staticmethod
+    def construct_data(eopatch, features):
+        """
+        :param eopatch: EOPatch with all features that will be used
+        :param features: Features for clustering
+        :return: array of vectors constructed from the features listed
+        """
+
+        feature = EOTask._parse_features(features)
+        data = None
+        for i in feature:
+            if data is None:
+                data = eopatch.data_timeless[i[1]]
```

Comment:
Tha tasks seems to assume that clusters will be always determined using `data_timeless` feature. If this is the case, than make this explicit in the task constructor:

```
self.features  = self._parse_features(features, default_feature_type=FeatureType.DATA_TIMELESS)
```